### ISSUE TITLE
*potential* multiplication error in 'unsigned int' before it is converted to 'int64_t'

### ISSUE THREAD

**Issue by monsterd0n** [stellar/stellar-core#1808]
I have been reading the source of stellar trying to look for hidden bugs and I may have found one that could potentially be a bug (unless if someone understands the code differently than I do after looking at this).
I could be wrong though
https://github.com/stellar/stellar-core/blob/29d4b1a16289161822251dd91d589ebe943b3f46/src/ledger/LedgerManagerImpl.cpp#L339

**Comment by MonsieurNicolas** [comment]
Yes this is a bug that was fixed in protocol version 9: if you notice the conditional on the statement and the fix in the "else".

This was described in the release notes for 9.0.0
